synapse/storage/databases/main/pusher.py

```diff
@@ -595,8 +592,6 @@ async def add_pusher(
         device_id: Optional[str] = None,
     ) -> None:
         async with self._pushers_id_gen.get_next() as stream_id:
-            # no need to lock because `pushers` has a unique key on
-            # (app_id, pushkey, user_name) so simple_upsert will retry
```

Comment:
This seems to have been true since migration 40, but not when the table was added (migration 12) nor when it was altered (migration 15)